scipy/optimize/minpack.py

```diff
@@ -510,28 +523,35 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         values will all be 1 (if the number of parameters for the function
         can be determined using introspection, otherwise a ValueError
         is raised).
-    sigma : None or M-length sequence, optional
-        If not None, the uncertainties in the ydata array. These are used as
-        weights in the least-squares problem
-        i.e. minimising ``np.sum( ((f(xdata, *popt) - ydata) / sigma)**2 )``
-        If None, the uncertainties are assumed to be 1.
+    sigma : None or M-length sequence or MxM array, optional
+        Determines the uncertainty in `ydata`. If we define residuals as
+        ``e = ydata - f(xdata, *popt)``, then
+
+            - If `sigma` is 1-d, it determines standard deviations of residuals `e`
```

Comment:
Sounds good. I was a little bit hesitant in my previous commit, as sigma is
often determined from resampling methods without recourse to a specific
model f(xdata, *popt). Your modification sounds good. see 34ee873





On Sun, Aug 21, 2016 at 12:52 AM Nikolay Mayorov <notifications@github.com>
wrote:

> In scipy/optimize/minpack.py
> <https://github.com/scipy/scipy/pull/6493#discussion_r75580690>:
>
> > @@ -510,28 +523,35 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
> >          values will all be 1 (if the number of parameters for the function
> >          can be determined using introspection, otherwise a ValueError
> >          is raised).
> > -    sigma : None or M-length sequence, optional
> > -        If not None, the uncertainties in the ydata array. These are used as
> > -        weights in the least-squares problem
> > -        i.e. minimising ``np.sum( ((f(xdata, *popt) - ydata) / sigma)**2 )``
> > -        If None, the uncertainties are assumed to be 1.
> > +    sigma : None or M-length sequence or MxM array, optional
> > +        Determines the uncertainty in `ydata`. If we define residuals as
> > +        ``e = ydata - f(xdata, *popt)``, then
> > +
> > +            - If `sigma` is 1-d, it determines standard deviations of residuals `e`
>
> I guess it's not totally correct to call sigma as standard deviations of
> *residuals*.
>
> How about:
>
> Determines the uncertainty in `ydata`. If we define residuals as
> ``e = ydata - f(xdata, *popt)``, then the interpretation of `sigma`
> depends on its number of dimensions:
>
> 1-d `sigma` contains values of standard deviations of errors in `ydata`.
> The optimized function is ``chisq = sum((e / sigma) ** 2)``.
>
> 2-d `sigma` is the covariance matrix of errors in `ydata`.
> The optimized function is ``chisq = e.T.dot(inv(sigma)).dot(e)``.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scipy/scipy/pull/6493/files/25c3b367184a98db32048ee83d89b0f21ddd1c8a#r75580690>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAb4jQldjAaauzGlOJaO3Ubj_KHuG6zFks5qhyLLgaJpZM4JmQ86>
> .
>